### PR TITLE
test(esbuild): Add test to check for compatibility with `inject` option

### PR DESCRIPTION
**Pull request by lforst**
Ref: https://github.com/getsentry/sentry-javascript-bundler-plugins/issues/413

Verify that we are not stomping the `inject` option somehow.

### CODE REVIEW

File: packages/integration-tests/fixtures/esbuild-inject-compat/input/inject.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const process = { cwd: () => "", env: {} };
```
Comment (mydea) [inline packages/integration-tests/fixtures/esbuild-inject-compat/input/inject.ts:1]: Is the idea that this _overwrites_ `process.env.FOO`? Where would that be set?

Comment (lforst) [reply packages/integration-tests/fixtures/esbuild-inject-compat/input/inject.ts:1]: I could make the test a bit clearer actually...

Comment (lforst) [reply packages/integration-tests/fixtures/esbuild-inject-compat/input/inject.ts]: Adjusted here [d32130e](https://github.com/getsentry/sentry-javascript-bundler-plugins/pull/416/commits/d32130e5ad6bac1f6b64d7ae6b681eb40cadefb8)

**Review by mydea — APPROVED** [review]

**Comment by lforst** [comment]
Whoops this fails. But this is fixed with https://github.com/getsentry/sentry-javascript-bundler-plugins/pull/417 🤦

**Pull request closed without merging**